### PR TITLE
Player template adjustments

### PR DESCRIPTION
**Pull request by geekygnr**
Resolves
- #33 
- #36 

Some processing on the the render array will need to happen to add some of the meta data that is specific to the site. Only meta data that doesn't change on a per site basis has been added.

### CODE REVIEW

**Comment by geekygnr** [comment]
I don't think it makes sense to do the refactoring it suggests. What do you think @deviantintegral?

File: README.md [hunk 1]
```diff
@@ -196,3 +197,13 @@ machine name is `mpx_lullabot`, add the following to `settings.php`:
 
 $config['media_mpx.media_mpx_user.mpx_lullabot']['password'] = 'SECRET';
 ```
+## Interacting with the Video Player
```
Comment (deviantintegral) [inline README.md:200]: 👍

File: src/Element/MediaMpxIframeWrapper.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Drupal\media_mpx\Element;
+
+use Drupal\Core\Render\Element\RenderElement;
+use Drupal\Core\Template\Attribute;
+
+/**
+ * Providers an element design for wrapping media iframes.
```
Comment (deviantintegral) [inline src/Element/MediaMpxIframeWrapper.php]: Typo in `Providers`

File: src/Element/MediaMpxIframeWrapper.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   *
+   * @param array $element
+   *   An element array before being processed.
+   *
+   * @return array
+   *   The processed and renderable element.
+   */
+  public static function preRenderInlineFrameEmbed(array $element) {
+    if (is_array($element['#attributes'])) {
+      $element['#attributes'] = new Attribute($element['#attributes']);
```
Comment (deviantintegral) [inline src/Element/MediaMpxIframeWrapper.php]: Ooo interesting, I thought core handled `#attributes` for you automatically this way.

File: src/Plugin/Field/FieldFormatter/PlayerFormatter.php [hunk 4]
```diff
@@ -244,23 +247,98 @@ protected function renderIframes(FieldItemListInterface $items, Player $player,
...
+      '#type' => 'media_mpx_iframe_wrapper',
+      '#attributes' => [
+        'class' => [
+          'mpx-iframe-wrapper',
+        ],
+      ],
+      '#meta' => [
+        'name' => $entity->label(),
+        'thumbnailUrl' => $thumbnail_url,
+        'uploadDate' => $mpx_media->getAvailableDate()->format(DATE_ISO8601),
```
Comment (deviantintegral) [inline src/Plugin/Field/FieldFormatter/PlayerFormatter.php:290]: Yay, glad this works.

File: src/Plugin/Field/FieldFormatter/PlayerFormatter.php [hunk 5]
```diff
@@ -244,23 +247,98 @@ protected function renderIframes(FieldItemListInterface $items, Player $player,
...
+      ],
+      '#content' => $this->buildPlayer($source_plugin, $player, $mpx_media),
+    ];
+    $this->addMediaFileDetails($element, $mpx_media);
+
+    return $element;
+  }
+
+  /**
+   * Adds the video specific details to schema.org meta data.
```
Comment (deviantintegral) [inline src/Plugin/Field/FieldFormatter/PlayerFormatter.php]: `metadata` is usually one word.

I'm going to note that we pull from the first file.

File: tests/src/FunctionalJavascript/PlayerFormatterTest.php [hunk 6]
```diff
@@ -113,7 +113,7 @@ public function testViewPlayerFormatter() {
     $this->getSession()->getPage()->fillField('Name', $this->randomString());
     $this->getSession()->getPage()->fillField('mpx Media', 'http://data.media.theplatform.com/media/data/Media/2602559');
     $this->click('#edit-submit');
-    $this->assertSession()->responseContains('<iframe src="https://player.theplatform.com/p/public-id/DemoPlayer/select/media/Zy1n9JQPy6hw"></iframe>');
+    $this->assertSession()->responseContains('<iframe class="mpx-player mpx-player-account--' . $account->id() . '" src="https://player.theplatform.com/p/public-id/DemoPlayer/select/media/Zy1n9JQPy6hw"></iframe>');
```
Comment (deviantintegral) [inline tests/src/FunctionalJavascript/PlayerFormatterTest.php:116]: You got this working, nice!

**Review by deviantintegral — COMMENTED** [review]
This looks great. I'm going to do some final touchups so we can get this finished in our current sprint.

File: src/Element/MediaMpxIframeWrapper.php (hunk 3 above)
Comment (deviantintegral) [reply src/Element/MediaMpxIframeWrapper.php]: Hah, you got this from my code I wrote back in April, so clearly I ran into this already.

**Comment by deviantintegral** [comment]
I think it's worth pulling a base class for the render elements, given that it's simple (5 minutes, not the hour code climate estimates) and that the only difference is the theme value.

**Pull request merged**